docs/latest/design/egctl.md

```diff
@@ -0,0 +1,73 @@
+# Introduce egctl
+
+## Motivation
+
+EG should provide a command line tool with following capabilities:
+
+- One-step installation/uninstallation experience
+- Collect configuration from envoy proxy and gateway
+- Update admin options such as changing log level
+- Analyse system configuration to diagnose any issues in envoy gateway
+
+This tool is named `egctl`.
+
+## Syntax
+
+Use the following syntax to run `egctl` commands from your terminal window:
+
+```console
+egctl [command] [entity] [name] [flags]
+```
+
+where `command`, `name`, and `flags` are:
+
+* `command`: Specifies the operation that you want to perform on one or more resources,
+  for example `install`, `uninstall`, `config`, `version`.
+
+* `entity`: Specifies the entity the operation is being performed on such as `envoy-proxy` or `envoy-gateway`.
+
+* `name`: Specifies the name of the specified instance. 
+
+* `flags`: Specifies optional flags. For example, you can use the `-c` or `--config` flags to specify the values for installing.
+
+If you need help, run `kubectl help` from the terminal window.
+
+## Operation
+
+The following table includes short descriptions and the general syntax for all the `egctl` operations:
+
+| Operation   | Syntax                           | Description                                                                 |
+| ----------- | -------------------------------- | --------------------------------------------------------------------------- |
+| `version`   | `egctl version`                  | Prints out build version information.                                       |
+| `install`   | `egctl install -c CUSTOMPROFILE` | Install or reconfigure EG on a cluster.                                     |
```

Comment:
I believe `-c` should reference a config file in [EnvoyGateway](https://github.com/envoyproxy/gateway/blob/main/api/config/v1alpha1/envoygateway_types.go) format.

Reply:
Actually, `-c` is alias for `--kubeconfig` in kubernetes, should `EG` keep constaint with others?

Reply:
Yes, if kubectl uses `-c` to specify the kubeconfig, then `-c` should be used to specify the egconfig file. The egconfig file will be similar to the kubeconfig, i.e. contain EG endpoints, credentials, etc. Let's pick a different short name for specifying the EnvoyGateway/EnvoyProxy config used with the `install` subcommand.